### ISSUE TITLE
question about validation loss

### ISSUE THREAD

**Issue by 2hyes** [eriklindernoren/PyTorch-YOLOv3#632]
hi. 
I want to minimize validation loss during training. 
Where can I find the code about validation loss ?

**Comment by Flova** [comment]
We currently feature multiple metrics during the evaluation (mAP, f1, precision, recall). The loss computed in the same way as during the training is not a part of them.

To compute it you would need to pass the targets to the model execution in the `test.py` file. It would return the loss and the predictions in return.

```
 with torch.no_grad():
            outputs = model(imgs)
            outputs = non_max_suppression(outputs, conf_thres=conf_thres, nms_thres=nms_thres)
```

to 

```
 with torch.no_grad():
            outputs, loss = model(imgs, targets)
            outputs = non_max_suppression(outputs, conf_thres=conf_thres, nms_thres=nms_thres)
```

This would be computed for every batch you therefore need to reduce it and convert it to a tensor that is accessible by the cpu.
